index.bs

```diff
@@ -6419,8 +6465,7 @@ as follows. The algorithm takes a |value|, a |key| and a |keyPath|.
 4. For each remaining |identifier| in |identifiers|, run these
     substeps:
 
-    1. If |value| is not an [=Object=] object or an [=Array=]
-        object, then [=throw=] a "{{DataError}}" {{DOMException}}.
+    1. Assert: |value| is an [=Object=] or an [=Array=].
```

Comment:
Right. "Assert" is just a guide to implementers to help with the readability of algorithms. Throwing required breaking out the steps separately, as @brettz9 clarifies.